osquery/tables/system/darwin/kernel_panics.cpp

```diff
@@ -0,0 +1,137 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <boost/algorithm/string/predicate.hpp>
+#include <boost/algorithm/string/replace.hpp>
+#include <boost/algorithm/string/trim.hpp>
+#include <boost/regex.hpp>
+
+#include <osquery/filesystem.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/conversions.h"
+#include "osquery/tables/system/system_utils.h"
+
+namespace fs = boost::filesystem;
+namespace alg = boost::algorithm;
+
+namespace osquery {
+namespace tables {
+
+/// Location of the kernel panic crash logs in OS X
+const std::string kDiagnosticReportsPath = "/Library/Logs/DiagnosticReports";
+
+/// List of all register values we wish to catch
+const std::set<std::string> kRegisters = {
+    "CR0", "RAX", "RSP", "R8", "R12", "RFL",
+};
+
+/// List of the days of the Week, used to grab our timestamp.
+const std::set<std::string> kDays = {"Mon", "Tue", "Wed", "Thu", "Fri"};
+
+/// Map of the values we currently parse out of the log file
+const std::map<std::string, std::string> kKernelPanicKeys = {
+    {"dependency", "kext_dependency"},
+    {"BSD process name corresponding to current thread", "name"},
+    {"System model name", "system_model"},
+    {"System uptime in nanoseconds", "uptime"},
+};
+
+void readKernelPanic(const std::string& appLog, Row& r) {
+  r["path"] = appLog;
+  std::string content;
+
+  if (!readFile(appLog, content).ok()) {
+    return;
+  }
+
+  boost::regex rxSpaces("\\s+");
+  auto lines = osquery::split(content, "\n");
+  for (auto it = lines.begin(); it != lines.end(); it++) {
+    auto line = *it;
+    boost::trim(line);
+
+    auto toks = osquery::split(line, ":");
+    if (toks.size() == 0) {
+      continue;
+    }
+
+    auto timeTokens = osquery::split(toks[0], " ");
+    if (timeTokens.size() >= 1 && kDays.count(timeTokens[0]) > 0) {
+      r["time"] = line;
+    }
+
+    if (kRegisters.count(toks[0]) > 0) {
+      auto registerTokens = osquery::split(line, ",");
+      if (registerTokens.size() == 0) {
+        continue;
+      }
+      std::string regLine = "";
+      for (auto& tok_ : registerTokens) {
+        auto regHolder = osquery::split(tok_, ":");
+        if (regHolder.size() != 2) {
+          continue;
+        }
+        auto reg = regHolder[0];
```

Comment:
Do you intend to make copies of these values here? You can `auto&` and then `std::move` them into the `regLine` holder.